docs/literate/src/files/behind_the_scenes_simulation_setup.jl

```diff
@@ -0,0 +1,240 @@
+#src # Behind the scenes of a simulation setup
+
+# This tutorial will guide you through a simple Trixi.jl setup ("elixir"), giving an overview of what
+# happens in the background during the initialization of a simulation. While the setup described herein
+# does not cover all details, it is involves relatively stable parts of Trixi.jl that are unlikely
+# to undergo significant changes in the near future. The goal is to clarify some of the more
+# fundamental, *technical* concepts that are applicable to a variety of (also more complex)
+# configurations.
```

Comment:
Maybe state explicitly already here (or a little further down) that Trixi.jl follows the concept of method of lines, i.e. the partial differential equation is reduced to an ordinary differential equation by first discretizing space ("semidiscretization") and in the end solve solve an ordinary differential equation? This gives the reader a very basic guideline of the following steps.